uploader/s3_test.go

```diff
@@ -0,0 +1,25 @@
+package uploader
+
+import (
+	"io"
+	"strings"
+	"testing"
+)
+
+func TestCountingReader_Read(t *testing.T) {
+	const data = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	reader := strings.NewReader(data)
+	countingReader := &CountingReader{reader: reader}
+
+	buffer := make([]byte, 10)
+	for {
+		_, err := countingReader.Read(buffer)
```

Comment:
You can use:
* https://golang.org/pkg/io/ioutil/#ReadAll
* https://golang.org/pkg/io/#Copy with `ioutil.discard`

There's also a library which makes asserting on these errors easier: https://github.com/stretchr/testify

You can do something like `assert.Equal(t, len(data), n)`